tools/build-tools/src/test/typeValidator/classValidator.spec.ts

```diff
@@ -49,92 +49,503 @@ describe("Class", () => {
+
+            const classNew =
+            `
+            export class TestClass {
+                private _testProperty = "sussy baka";
+                get testProperty() { return this._testProperty; }
+            }
+            `;
+
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.major);
+        });
+
+        // setters are treated the same as public methods
+        // adding/removing one is a breaking change
+        it("added a setter", () => {
+            const classOld =
+            `
+            export class TestClass {
+                private _testProperty = "sussy baka";
+            }
+            `;
+
+            const classNew =
+            `
+            export class TestClass {
+                private _testProperty = "sussy baka";
+                set setTestProperty(newTestProperty: string) {
+                    this._testProperty = newTestProperty;
+                }
+            }
+            `;
 
-        let increment = checkIncrement(classOld, classNew);
-        assert(increment == BreakingIncrement.major);
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.major);
+        });
+
+        // adding a new required method parameter is a breaking change
+        it("added a new required method param", () => {
+            const classOld =
+            `
+            export class TestClass {
+                public testMethod() { return false; }
+            }
+            `;
+
+            const classNew =
+            `
+            export class TestClass {
+                public testMethod(newProp: boolean) { return false; }
+            }
+            `;
+
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.major);
+        });
+
+        // scenario: added new default param (method signature changed but won't break existing code)
+        // expected result: minor breaking change
+        it("new default value added", () => {
+            const classOld =
+            `
+            export class TestClass {
+                public testMethod() { return "afg"; }
+            }
+            `;
+
+            const classNew =
+            `
+            export class TestClass {
+                public testMethod(param1 : string = "afg") { return "afg"; }
+            }
+            `;
+
+            let increment = checkIncrement(classOld, classNew);
+            assert.strictEqual(increment, BreakingIncrement.minor);
+        });
+
+        // scenario: new version changes the param to a different type.
+        // expected result: A major breaking change
+        it("default value type changed", () => {
+            const classOld =
+            `
+            export class TestClass {
+                public testMethod(param1 : string = "afg") { return "afg"; }
+            }
```

Comment:
Will this also work with classes? I.e. if you used classes instead of primitive types, would that be testing something different?

Reply:
for classes what happens is that the types are replaced and checked separately (which is how transitive breaks are checked).  those transitivity tests haven't been written yet, but changing from one class type to another is already covered incidentally here.  if you're talking about default values applied to the same class type, i don't think that's ever a breaking change